src/stronghold/mod.rs

```diff
@@ -112,16 +113,12 @@ impl StrongholdAdapterBuilder {
     /// Use an user-input password string to derive a key to use Stronghold.
     pub fn password(mut self, password: &str) -> Self {
         // Note that derive_builder always adds another layer of Option<T>.
-        self.key = Some(Arc::new(Mutex::new(Some(self::common::derive_key_from_password(
-            password,
-        )))));
+        self.key = Some(self::common::derive_key_from_password(password));
```

Comment:
This is insecure. Even though a KDF will be used to store the secret inside an `Option`, the secret itself will remain in memory. I would recommend to re-use the `KeyProvider` data structure to store the key inside `StrongholdAdapter`. 
Further down below, the `KeyProvider` is already being used.